### PR TITLE
Bump konnectivity to v0.0.25

### PR DESCRIPTION
**Pull request by cheftako**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Bump konnectivity network proxy to v0.0.25.
Includes fixes for a few resource leaks.
Adds better logging for debugging.
Moves to golang1.17.
Adds additional keepalives.
Fixes HTTP-CONNECT goroutine leak. 

#### Which issue(s) this PR fixes:

Fixes # N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,6 +39,7 @@ require (
 	github.com/emicklei/go-restful v2.9.5+incompatible
 	github.com/evanphx/json-patch v4.12.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
+	github.com/go-openapi/spec v0.20.4
```
Comment (liggitt) [inline go.mod:42]: this is spurious... `make clean generated_files` should drop the need for this

File: go.mod [hunk 2]
```diff
@@ -516,7 +518,7 @@ replace (
 	modernc.org/strutil => modernc.org/strutil v1.0.0
 	modernc.org/xc => modernc.org/xc v1.0.0
 	rsc.io/pdf => rsc.io/pdf v0.1.1
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.23
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.25
```
Comment (liggitt) [inline go.mod:521]: no changes in vendor... this diff didn't include any client files, I guess?

Comment (cheftako) [reply go.mod:521]: Went and explicitly checked. The last konnectivity-client fix appears to be August 2. However I like to just keep these 3 pieces in sync when we update.

File: go.mod (hunk 1 above)
Comment (cheftako) [reply go.mod]: Done

Comment (liggitt) [reply go.mod]: still seeing other openapi changes in the go.mod files ... they should be dropped from the PR

File: go.mod [hunk 3]
```diff
@@ -452,7 +452,7 @@ replace (
 	golang.org/x/sync => golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
 	golang.org/x/sys => golang.org/x/sys v0.0.0-20210820121016-41cdb8703e55
 	golang.org/x/term => golang.org/x/term v0.0.0-20210615171337-6886f2dfbf5b
-	golang.org/x/text => golang.org/x/text v0.3.6
+	golang.org/x/text => golang.org/x/text v0.3.7
```
Comment (liggitt) [inline go.mod:455]: sorry to keep pressing on this, but there shouldn't be any go.mod changes other than `sigs.k8s.io/apiserver-network-proxy/konnectivity-client`, right?

File: go.mod (hunk 1 above)
Comment (cheftako) [reply go.mod]: clean dropped it from go.su, but not go.mod. Removed it manually and confirmed the steps didn't re-add it. Should be good now.

**Pull request closed without merging**